src/Microsoft.OData.Edm/EdmModelVisitor.cs

```diff
@@ -843,7 +843,7 @@ protected virtual void ProcessOperation(IEdmOperation operation)
         {
             if (operation.ReturnType != null)
             {
-                this.VisitTypeReference(operation.ReturnType);
+                this.VisitOperationReturnType(operation.ReturnType as IEdmOperationReturnType);
             }

```

Comment:
For backward compatibility, it seems like we should attempt to cast operation.ReturnType to IEdmOperationReturnType and, if that failed, just call VisitTypeReference.

Reply:
Basically, it looks safe because we can make sure the "operation.ReturnType" will return IEdmOperationReturnType object for ever with this change. We make that in the "EdmOperation" constructor method.

But,  it's also ok to keep backward compatibility to check the object type first. So, #Resolved.